src/tyr/route_serializer_valhalla.cc

```diff
@@ -530,6 +530,43 @@ void legs(const valhalla::Api& api, int route_index, rapidjson::writer_wrapper_t
       writer.end_array(); // elevation
     }
 
+    // Does the user want admin info?
+    if (api.options().admin_crossings() && trip_leg_itr->admin_size() > 0) {
```

Comment:
so maybe we only want this when its more than 1 meaning an actual crossing occured, you left one and went to at least one other?

Reply:
We should always get the `admins` array even if it's only one (it's never going to be less than that), and only get the crossings when there are at least two admins along the route. Updated the code to reflect that.